### PR TITLE
Add traffic allocation support

### PR DESCRIPTION
**Pull request by maksimf**

### CODE REVIEW

**Comment by maksimf** [comment]
@patricioe This PR can be merged, what version should I use? 3.3.0?

File: lib/splitclient-rb/clients/split_client.rb [hunk 1]
```diff
@@ -77,34 +77,42 @@ def get_treatment(key, split_name, attributes = nil, split_data = nil, store_imp
...
-        end
+        # Disable impressions if @config.impressions_queue_size == -1
+        split && store_impression(split_name, matching_key, bucketing_key, treatment_label_change_number, store_impressions)
 
         # Measure
         @adapter.metrics.time('sdk.get_treatment', latency)
       rescue StandardError => error
         @config.log_found_exception(__method__.to_s, error)
 
+        store_impression(split_name, matching_key, bucketing_key, treatment_label_change_number, store_impressions)
```
Comment (patricioe) [inline lib/splitclient-rb/clients/split_client.rb]: label should be exception here, no?

File: spec/engine_spec.rb [hunk 2]
```diff
@@ -357,6 +358,32 @@
...
       end
+
+      context 'traffic allocations' do
+        before do
+          stub_request(:get, 'https://sdk.split.io/api/splitChanges?since=-1')
+            .to_return(status: 200, body: traffic_allocation_json)
+        end
+
+        it 'returns expected treatment' do
+          expect(subject.get_treatment('01', 'Traffic_Allocation_UI')).to eq(SplitIoClient::Treatments::OFF)
```
Comment (patricioe) [inline spec/engine_spec.rb]: do we have a case where traffic allocation is 0% and check for default treatment? 
Also the opposite. I saw a test right underneath here but I'm not sure what the default treatment is.

File: spec/engine_spec.rb [hunk 3]
```diff
@@ -357,6 +358,32 @@
...
+          expect(subject.get_treatment('ab', 'Traffic_Allocation_UI')).to eq(SplitIoClient::Treatments::OFF)
+          expect(subject.get_treatment('00b0', 'Traffic_Allocation_UI')).to eq(SplitIoClient::Treatments::OFF)
+        end
+
+        it 'returns expected treatment when traffic alllocation < 100' do
+          expect(subject.get_treatment('01', 'Traffic_Allocation_UI3')).to eq(SplitIoClient::Treatments::OFF)
```
Comment (patricioe) [inline spec/engine_spec.rb]: definitely try a non default treatment. See comment above.

File: spec/test_data/splits/splits_traffic_allocation.json [hunk 4]
```diff
@@ -0,0 +1,281 @@
+{
+    "splits": [
+        {
+            "trafficTypeName": "account",
+            "name": "Traffic_Allocation_UI",
+            "trafficAllocation": 100,
+            "trafficAllocationSeed": null,
```
Comment (patricioe) [inline spec/test_data/splits/splits_traffic_allocation.json]: all the seeds are  null. Intentional?

File: lib/splitclient-rb/clients/split_client.rb (hunk 1 above)
Comment (maksimf) [inline lib/splitclient-rb/clients/split_client.rb]: Yep, fixed

File: spec/engine_spec.rb (hunk 2 above)
Comment (maksimf) [inline spec/engine_spec.rb]: Done

File: spec/engine_spec.rb (hunk 3 above)
Comment (maksimf) [inline spec/engine_spec.rb]: Done

**Comment by maksimf** [comment]
@patricioe Fixed, I've used version `3.3.0` for this version, is it ok?

File: CHANGES.txt [hunk 5]
```diff
@@ -1,3 +1,6 @@
+3.3.0
+- Add support for murmur hash
```
Comment (patricioe) [inline CHANGES.txt]: did you mean to say traffic allocation?

Comment (maksimf) [inline CHANGES.txt]: Sure, fixed

File: lib/engine/models/label.rb [hunk 6]
```diff
@@ -2,10 +2,11 @@ module SplitIoClient
...
       class Label
-        ARCHIVED = 'archived'
-        NO_RULE_MATCHED = 'no rule matched'
-        EXCEPTION = 'exception'
-        KILLED = 'killed'
+        ARCHIVED = 'archived'.freeze
+        NO_RULE_MATCHED = 'no rule matched'.freeze
+        EXCEPTION = 'exception'.freeze
+        KILLED = 'killed'.freeze
+        NOT_IN_SPLIT = 'not included in Split'.freeze
```
Comment (adilaijaz) [inline lib/engine/models/label.rb]: 'not in split' instead of 'not included in Split'

Comment (maksimf) [inline lib/engine/models/label.rb]: Fixed, thank you

**Pull request merged**